src/methods/to.js

```diff
@@ -669,6 +669,17 @@ export class ToImplementation {
         let resultType = result.getType();
 
         // ii. If Type(result) is not Object, return result.
+        if (resultType === Value) {
+          invariant(result instanceof AbstractValue);
+          let error = new CompilerDiagnostic(
+            `${name} might return either an object or primitive`,
+            realm.currentLocation,
+            "PP0028",
```

Comment:
This error does not seem much more informative than PP0002. Perhaps the latter will do nicely in this situation?